experiments/download_ssl4eo.py

```diff
@@ -396,23 +393,20 @@ def get_random_patches_grid(
     # avoid strong overlap
     try:
         new_coord = (coords[0], coords[1])
-        gridIndex = (math.floor(new_coord[0] + 180), math.floor(new_coord[1] + 90))
-
-        if gridIndex not in grid_dict.keys():
-            grid_dict[gridIndex] = {new_coord}
-        else:
-            for coord in grid_dict[gridIndex]:
-                distance = np.sqrt(
-                    sampler.deg2km(abs(new_coord[0] - coord[0])) ** 2
-                    + sampler.deg2km(abs(new_coord[1] - coord[1])) ** 2
-                )
-                if distance < (1.5 * radius / 1000):
-                    raise OverlapError
-            grid_dict[gridIndex].add(new_coord)
+        for i in rtree_obj.nearest(new_coord, num_results=1, objects=True):
+            distance = np.sqrt(
+                sampler.deg2km(abs(new_coord[0] - i.bbox[2])) ** 2
+                + sampler.deg2km(abs(new_coord[1] - i.bbox[3])) ** 2
+            )
+            if distance < (1.5 * radius / 1000):
+                raise OverlapError
```

Comment:
There's a much easier way to do this:
```pycon
>>> from rtree.index import Index
>>> idx = Index()
>>> idx.insert(0, (0, 0, 1, 1))
>>> list(idx.intersection((0, 0, 1, 1)))
[0]
>>> list(idx.intersection((2, 2, 3, 3)))
[]
```
Just check if the new bbox intersects with any existing bboxes in the R-tree. R-trees are designed to store boxes, not points (although you can also store points). The above example requires you to store the bounding box of the patch you want to store.